### PR TITLE
Attestation workflow

### PR DESCRIPTION
**Pull request by alpe**
* New CLI commands and query handlers
* Dev scripts for manual tests
* Some code cleanup and documentation

### CODE REVIEW

File: module/x/peggy/client/cli/oracle_tx.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+func CmdSendETHMultiSigRequest(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "multisig-update [eth chain id] [eth contract address] [nonce]",
+		Short: "Submit an eth event observed by an orchestrator",
```
Comment (jkilpatr) [inline module/x/peggy/client/cli/oracle_tx.go:126]: can we make the description more explicit. I think this is for updating the multisig set but I'm not 100% sure which I would say means the description is too vague.

File: module/x/peggy/client/cli/oracle_tx.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+func CmdSendETHWithdrawalRequest(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "withdrawal [eth chain id] [eth contract address] [nonce]",
+		Short: "Submit an eth event observed by an orchestrator",
```
Comment (jkilpatr) [inline module/x/peggy/client/cli/oracle_tx.go:91]: description needs more detail. this is for a tx batch?

File: module/x/peggy/client/cli/oracle_tx.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		CmdSendETHMultiSigRequest(cdc),
+	)...)
+
+	return testingTxCmd
+}
+
+func CmdSendETHDepositRequest(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "deposit [eth chain id] [eth contract address] [nonce] [cosmos receiver] [amount] [eth erc20 symbol] [eth erc20 contract addr] [eth sender address]",
+		Short: "Submit an eth event observed by an orchestrator",
```
Comment (jkilpatr) [inline module/x/peggy/client/cli/oracle_tx.go:40]: this is very clearly for a deposit, still think it needs better description.

File: module/x/peggy/client/cli/oracle_tx.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+			}
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+// todo: we mix nonces as int64 and base64 bytes at the moment
```
Comment (jkilpatr) [inline module/x/peggy/client/cli/oracle_tx.go:158]: why is it mixed?

File: module/x/peggy/types/pool.go [hunk 5]
```diff
@@ -12,18 +12,42 @@ import (
...
-	Amount      sdk.Coin       `json:"send"`
-	BridgeFee   sdk.Coin       `json:"bridge_fee"`
-	//TokenContractAddress string         `json:"bridge_contract_address"` // todo: do we need this?
+	Sender      sdk.AccAddress  `json:"sender"`
+	DestAddress EthereumAddress `json:"dest_address"`
+	Amount      sdk.Coin        `json:"send"`
+	BridgeFee   sdk.Coin        `json:"bridge_fee"`
 }
 
 // BridgedDenominator contains bridged token details
```
Comment (jkilpatr) [inline module/x/peggy/types/pool.go:21]: some more details about how this is built in the comments would be nice.

**Review by jkilpatr — APPROVED** [review]

File: module/x/peggy/client/cli/oracle_tx.go (hunk 4 above)
Comment (alpe) [reply module/x/peggy/client/cli/oracle_tx.go:158]: This is related to https://github.com/althea-net/peggy/issues/48 . For generalization I was moving towards a []byte representation. That would be more flexible. After MVP we should revisit the different nonce types.

File: module/x/peggy/client/cli/oracle_tx.go (hunk 1 above)
Comment (alpe) [reply module/x/peggy/client/cli/oracle_tx.go:126]: Fair comments on the doc. There was definitively no priority to make this a good UX 🙈

**Comment by alpe** [comment]
I will address the comments in PR #54

**Pull request merged**